Unity-Technologies/ScriptableRenderPipeline
# [Shader Graph] Update Error Messaging for Sub Graph Previews

### Purpose of this PR
Why is this PR needed, what hard problem is it solving/fixing? 
The preview renderer cannot handle certain data types implicitly. The Sub Graph output node allows these data types as output ports, but enters an error state if the first port (used for rendering previews) is an invalid data type. Additionally, the existing validation for empty ports actually hit a null ref before presenting the error message. 
This fixes the existing validation for empty port lists and display the correct error message. 
This also adds an extra validation path for invalid slot types. If the preview renderer is failing because of an incompatible data type, the node will validate and tell the user why. Additionally, docs have been updated to reflect this preview state and explicitly state supported data types.
It is expected that both of these states make the sub graph asset invalid. If the user tries to save while in this invalid state, errors will log in the console. If the user tries to use the subgraph in a main graph while in this state, the referenced Sub Graph node will throw an Invalid Asset error. 

[internal bug link only](https://fogbugz.unity3d.com/f/cases/1205973/)

---
### Testing status

**Manual Tests**: What did you do?
- Other: 
- Created a new sub graph asset. Put the slot list into several states of error (all invalid slot types as first input, no inputs at all)
- Checked the asset in main graphs while in error states to confirm "Invalid Asset" error is shown. 
- Put the graph back in a valid state and observed errors clearing in both the sub graph asset and any referenced graphs. 

![subGraphErrors](https://user-images.githubusercontent.com/39529353/77693594-d5594980-6f65-11ea-92d6-2617b16323ef.gif)

**Automated Tests**: What did you setup? (Add a screenshot or the reference image of the test please)

**Yamato**: (Select your branch):
https://yamato.prd.cds.internal.unity3d.com/jobs/78-ScriptableRenderPipeline

Any test projects to go with this to help reviewers?

---
### Comments to reviewers
Notes for the reviewers you have assigned.
Note to @Unity-Technologies/gfx-qa-bellevue , would you like this backported?
